### PR TITLE
Bump Gradle to version 4.1

### PR DESCRIPTION
**Pull request by JLLeitschuh**
## Overview

https://docs.gradle.org/4.1/release-notes.html

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests)
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -595,10 +595,10 @@ configure(rootProject) {
 
 	gitPublishCommit.dependsOn(prepareDocsForUploadToGhPages)
 	gitPublishCommit.dependsOn(createCurrentDocsFolder)
+}
 
-	task wrapper(type: Wrapper) {
```
Comment (JLLeitschuh) [inline build.gradle]: You really should only have this task on the root project.

File: build.gradle [hunk 2]
```diff
@@ -595,10 +595,10 @@ configure(rootProject) {
...
-		description = 'Generates gradlew and gradlew.bat scripts'
-		gradleVersion = '4.0.2'
-	}
-
+task wrapper(type: Wrapper) {
+	description = 'Generates gradlew and gradlew.bat scripts'
+	gradleVersion = '4.1'
+	distributionType = Wrapper.DistributionType.ALL
```
Comment (JLLeitschuh) [inline build.gradle]: Nice to have when writing/updating build logic.

Comment (sormuras) [inline build.gradle]: It was changed from "all" to "bin" some days ago via https://github.com/junit-team/junit5/commit/719127f615fce25a32fc4ed90514d8a748add9c5#diff-fd9d61e37d939c941e7a62077f38d62d -- please revert it as well.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Fri Jul 28 07:51:06 CEST 2017
+#Tue Aug 08 19:44:32 EDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.0.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (stefanbirkner) [inline gradle/wrapper/gradle-wrapper.properties]: I think it is better to use the `-bin.zip` because it is smaller.

File: build.gradle (hunk 1 above)
Comment (sormuras) [inline build.gradle]: True.

File: build.gradle (hunk 2 above)
Comment (JLLeitschuh) [inline build.gradle]: Sure, I can change it back. I personally work in gradle builds a lot at work and always finding the docs really helpful.

I'll explicitly set the distribution type to `BIN` with a mention of this comment chain.

File: build.gradle [hunk 4]
```diff
@@ -595,10 +595,14 @@ configure(rootProject) {
...
-		gradleVersion = '4.0.2'
-	}
-
+task wrapper(type: Wrapper) {
+	description = 'Generates gradlew and gradlew.bat scripts'
+	gradleVersion = '4.1'
+	/*
+	* Reasoning for using BIN:
+	* https://github.com/junit-team/junit5/pull/1012#discussion_r132061924
+	*/
```
Comment (JLLeitschuh) [inline build.gradle]: Commenting formatting seems weird but this is what spotless wants. so...

**Comment by JLLeitschuh** [comment]
I'm hoping that this unblocks #1001

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

**Pull request merged**